spec/unit/configurer_spec.rb

```diff
@@ -149,6 +149,37 @@
       expect(report.host).to eq('node_name_from_fact')
     end
 
+    it "should warn the user when the fact value length limits are exceeded" do
```

Comment:
We need to cover multiple cases here:
```
max_factvaluelength/max_toplevelfacts/max_factnamelength :
 -1
 0
 > 1
```

We need to have a variety of fact values(for https://github.com/puppetlabs/puppet/pull/8662/files#diff-51e52bffe1de00c153bde14dea475f15abdcb6a1fdaf2691633dad8238e654cfR159-R180), like nested hashes and arrays to reflect some real-world use cases.
eg. processors data:
```
processors => {
  cores => 1,
  count => 2,
  isa => "i386",
  models => [
    "CPU1 @ 2.80GHz",
    "CPU2 @ 2.80GHz"
  ],
  physicalcount => 4
```